bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: update minimum required Go version to 1.19.0

## Description

This PR updates the minimum required go version for KFM to 1.19.0. Go 1.17 is not supported anymore.

Changes:
* Update moq version
* Update golangci-lint version (and its corresponding GitHub action)
* Stop using `ioutil` in non-generated code as it is deprecated
* Fix missing autogeneration of some moq files
* Update different container image building files to use Go 1.19.0
* Stop using `-d` flag in `go get`as it is the default behavior now
* Updated minimum Go version requirements in README.md
* Update go.mod `go` directive to go 1.19

## Verification Steps
* Verify kas fleet manager can be built
* Verify `make generate` runs successfully
* Verify `make lint` runs successfully
* Verify `make verify` runs successfully
* Tests pass


## Checklist (Definition of Done)
